modules/highgui/src/window_cocoa.mm

```diff
@@ -646,6 +646,7 @@ CV_IMPL int cvWaitKey (int maxWait)
 
 CvRect cvGetWindowRect_COCOA( const char* name )
 {
+    NSAutoreleasePool *localpool = [[NSAutoreleasePool alloc] init];
```

Comment:
The reason I use autorelease pools directly is to maintain consistent coding style.

In this file, all 12 instances of autorelease pool is called directly:
https://github.com/opencv/opencv/blob/645fd3dffaa50e0b5dd011720048441986640d7e/modules/highgui/src/window_cocoa.mm#L194
https://github.com/opencv/opencv/blob/645fd3dffaa50e0b5dd011720048441986640d7e/modules/highgui/src/window_cocoa.mm#L211
https://github.com/opencv/opencv/blob/645fd3dffaa50e0b5dd011720048441986640d7e/modules/highgui/src/window_cocoa.mm#L223
...

IMO it would be more appropriate to create a new PR to clean up this file.

Reply:
@NekoAsakura Let's use `@autoreleasepool` for the new code as it's recommended in documentation.